pkg/storage/storage.go

```diff
@@ -80,69 +97,157 @@ func (p *storage) GetContainerMetrics(pods ...apitypes.NamespacedName) ([]api.Ti
 
 	timestamps := make([]api.TimeInfo, len(pods))
 	resMetrics := make([][]metrics.ContainerMetrics, len(pods))
+	for podIdx, pod := range pods {
+		contPoints, present := p.pods[pod]
+		if !present {
+			continue
+		}
 
-	for i, pod := range pods {
-		metricPoint, present := p.pods[pod]
+		_, present = p.prevPods[pod]
 		if !present {
 			continue
 		}
 
-		contMetrics := make([]metrics.ContainerMetrics, len(metricPoint.Containers))
-		var earliestTS *time.Time
-		for i, contPoint := range metricPoint.Containers {
-			contMetrics[i] = metrics.ContainerMetrics{
+		var (
+			contMetrics = make([]metrics.ContainerMetrics, len(contPoints))
```

Comment:
This will set both `cap` and `len` of `contMetrics` list to length of newer scrape. This can be a problem if there is a difference in containers between two scrapes. If there is a additional container in `p.pods[pod]` that is not present `p.prevPods[pod]` It will not report it's metric, but will still return list of size `p.pods[pod]` with one metric point empty. Empty result could be problematic in API so let's not do that.

This can be solved by using `make([]metrics.ContainerMetrics, 0, len(contPoints))` and `append` instead of setting value.

Context: https://riptutorial.com/go/example/3561/length-and-capacity

Reply:
Really good point, I didn't think about that.